lib/bolt/inventory.rb

```diff
@@ -243,5 +253,39 @@ def set_facts(tname, hash)
       end
     end
     private :set_facts
+
+    def add_node(grp, target, group_name, track = { 'all' => nil })
```

Comment:
I'd like to rename the variables to make this more readable. Something like

grp -> current_group
group_name -> desired_group
g -> child_group

track_item.name -> parent_name